publication-collector.js

```diff
@@ -32,13 +37,26 @@ PublicationCollector = class PublicationCollector extends EventEmitter {
     if (_.isFunction(args[args.length - 1])) {
       callback = args.pop();
     }
-    // adds a one time listener function for the "ready" event
-    this.once('ready', (collections) => {
+
+    const completeCollecting = (collections) => {
       if (_.isFunction(callback)) {
         callback(collections);
       }
-      // immediately stop the subscription
+
+      // stop the subscription
       this.stop();
+    };
+
+    // adds a one time listener function for the "ready" event
+    this.once('ready', (collections) => {
+      if (this.delayInMs) {
+        Meteor.setTimeout(() => {
+          completeCollecting(this._generateResponse());
```

Comment:
Mind putting a comment here explaining why we're calling `this._generateResponse()` instead of passing `collections`? I had to think for a while myself to grasp it :P